cql3/statements/cf_prop_defs.cc

```diff
@@ -290,6 +290,10 @@ void cf_prop_defs::apply_to_builder(schema_builder& builder, schema::extensions_
         cfm.caching(cachingOptions);
 #endif
 
+    // for extensions that are not altered, keep the old ones
+    auto& old_exts = builder.get_extensions();
+    schema_extensions.insert(old_exts.begin(), old_exts.end());
```

Comment:
What about extensions that *were* altered, does this do this alteration correctly?

I won't insist now, but a general request for this kind of patch: please add a test (or mention some pre-existing test which tests this). This will ensure that you test all the cases, and also tell the reviewer which cases you remembered to check, and which you forgot to check.

Reply:
I asked exactly the same thing @nyh. the extensions that were altered are already in schema_extensions and insert to a map fails if there is already an element under a given key so this code won't override altered extensions.

Reply:
Oh, I didn't see your question (I guess the code changed so your comment disappeared from this version).
I guessed this might be the explanation, but I'm still very unclear about the details: Is the only type of alteration completely replacing the old extension - will there never be a type of "merging" needed? What about *removing* an extension (e.g., previously you had CDC and now you want to remove it) - how will this work? 
If I saw tests that tested these cases, I wouldn't need to doubt that they are handled correctly, but without such tests, I have to raise doubts :-(

Reply:
Those are fair questions. Let's wait for @kbr- to answer them.

Reply:
>  Is the only type of alteration completely replacing the old extension - will there never be a type of "merging" needed?

We do the same thing with extension options (such as `cdc`) as with other table options (`compression`, `compaction`, etc.) do: when you set the option to something, you completely replace what was before. E.g.:
```
alter table ... with compression = {...}
```
replaces the `compression` option that was used before with whatever's in `{...}`.

We could add some merging, but that's a new feature which could be implemented in the future if people express the desire to have it.

> What about removing an extension (e.g., previously you had CDC and now you want to remove it) - how will this work?

You can't remove `compression`, you can't remove `compaction` (whatever you set it to, `describe table` will show it), and similarly, you won't be able to remove `cdc`. But you can disable cdc by e.g. using `with cdc = {}`.

I'll try adding some tests tomorrow.